java/src/main/java/com/google/appengine/tools/pipeline/di/AppEngineBackendModule.java

```diff
@@ -30,7 +30,7 @@ AppEngineBackEnd.Options appEngineBackEndOptions(AppEngineBackEnd pipelineBackEn
   @Module
   interface Bindings {
 
-    @Binds
+    @Binds @StepExecutionScoped
```

Comment:
all @Binds/@Provides methods seem to need explicit scopes.  scope annotations on classes ignored in these contexts.